File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        _sum = 0
        sum_indexes = {0: -1}

        for i, num in enumerate(nums):
            _sum = (_sum + num) % k

            if _sum in sum_indexes:
                if i - sum_indexes[_sum] >= 2:
                    return True
            else:
                sum_indexes[_sum] = i

        return False

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.go
func checkSubarraySum(nums []int, k int) bool {
	sum := 0
	sumIndexes := map[int]int{0: -1}

	for i, num := range nums {
		sum = (sum + num) % k

		if index, ok := sumIndexes[sum]; ok {
			if i-index >= 2 {
				return true
			}
		} else {
			sumIndexes[sum] = i
		}
	}

	return false
}